<a href="https://www.kaggle.com/code/animeshsinghiitm/heart-attack-analysis?scriptVersionId=115936681" target="_blank"><img align="left" alt="Kaggle" title="Open in Kaggle" src="https://kaggle.com/static/images/open-in-kaggle.svg"></a>

In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/heart-attack-analysis-prediction-dataset/o2Saturation.csv
/kaggle/input/heart-attack-analysis-prediction-dataset/heart.csv


# Author : Animesh Singh 
- Do upvote this notebook and appreciate.

**Let's load the data set**

In [2]:
heart=pd.read_csv("/kaggle/input/heart-attack-analysis-prediction-dataset/heart.csv")
o2sat=pd.read_csv("/kaggle/input/heart-attack-analysis-prediction-dataset/o2Saturation.csv")

**Now, Let's see top 5 rows to get an idea about all the features.**

In [3]:
heart.head()

age  sex  cp  trtbps  chol  fbs  restecg  thalachh  exng  oldpeak  slp  \
0   63    1   3     145   233    1        0       150     0      2.3    0   
1   37    1   2     130   250    0        1       187     0      3.5    0   
2   41    0   1     130   204    0        0       172     0      1.4    2   
3   56    1   1     120   236    0        1       178     0      0.8    2   
4   57    0   0     120   354    0        1       163     1      0.6    2   

   caa  thall  output  
0    0      1       1  
1    0      2       1  
2    0      2       1  
3    0      2       1  
4    0      2       1

In [4]:
o2sat.head()

98.6
0  98.6
1  98.6
2  98.6
3  98.1
4  97.5

***Let me introduce all the columns to you and there meanings***

   - **Age :** Age of the patient
   - **Sex :** Sex of the patient
   - **exang:** exercise induced angina (1 = yes; 0 = no)
   - **ca:** number of major vessels (0-3)
   - **cp :** Chest Pain type chest pain type
        Value 1: typical angina
        Value 2: atypical angina
        Value 3: non-anginal pain
        Value 4: asymptomatic
   - **trtbps :** resting blood pressure (in mm Hg)
   - **chol :** cholestoral in mg/dl fetched via BMI sensor
   - **fbs :** (fasting blood sugar > 120 mg/dl) (1 = true; 0 = false)
   - **rest_ecg :** resting electrocardiographic results
        Value 0: normal
        Value 1: having ST-T wave abnormality (T wave inversions and/or ST elevation or depression of > 0.05 mV)
        Value 2: showing probable or definite left ventricular hypertrophy by Estes' criteria
   - **thalach :** maximum heart rate achieved
   - **target :** 0= less chance of heart attack 1= more chance of heart attack

> Let's explore the dataset and see how many rows(data points) are in it. 

In [5]:
print("Number of rows in heart dataset: {row}".format(row=heart.shape))
print("Number of rows in o2Saturation dataset: {row_}".format(row_=o2sat.shape))

Number of rows in heart dataset: (303, 14)
Number of rows in o2Saturation dataset: (3585, 1)


**In this notebook we will work with heart dataset**
> To do so let's seperate the feature set from the target, in other words let's seperate the features and labels

In [6]:
data=heart.iloc[:,:-1]
label=heart.iloc[:,-1]

In [7]:
data.head()

age  sex  cp  trtbps  chol  fbs  restecg  thalachh  exng  oldpeak  slp  \
0   63    1   3     145   233    1        0       150     0      2.3    0   
1   37    1   2     130   250    0        1       187     0      3.5    0   
2   41    0   1     130   204    0        0       172     0      1.4    2   
3   56    1   1     120   236    0        1       178     0      0.8    2   
4   57    0   0     120   354    0        1       163     1      0.6    2   

   caa  thall  
0    0      1  
1    0      2  
2    0      2  
3    0      2  
4    0      2

In [8]:
label.head()

0    1
1    1
2    1
3    1
4    1
Name: output, dtype: int64

In [9]:
data.shape,label.shape

((303, 13), (303,))

**Now, let's see if our data has any missing value**

In [10]:
data.isnull().sum()

age         0
sex         0
cp          0
trtbps      0
chol        0
fbs         0
restecg     0
thalachh    0
exng        0
oldpeak     0
slp         0
caa         0
thall       0
dtype: int64

**It's a very clean dataset, generally in machine learning problem we don't get such clean data so we perform different data imputation techniques to fill up for that missing part  but unfortunately I can't show those techniques in this notebook, if you want to see few of those techniques you can go to my other notebooks where I had dealt with such data.**

____________________________________________________________________________________________________________________

**Now let's describe our data and see some statistics**

In [11]:
data.describe()

age         sex          cp      trtbps        chol         fbs  \
count  303.000000  303.000000  303.000000  303.000000  303.000000  303.000000   
mean    54.366337    0.683168    0.966997  131.623762  246.264026    0.148515   
std      9.082101    0.466011    1.032052   17.538143   51.830751    0.356198   
min     29.000000    0.000000    0.000000   94.000000  126.000000    0.000000   
25%     47.500000    0.000000    0.000000  120.000000  211.000000    0.000000   
50%     55.000000    1.000000    1.000000  130.000000  240.000000    0.000000   
75%     61.000000    1.000000    2.000000  140.000000  274.500000    0.000000   
max     77.000000    1.000000    3.000000  200.000000  564.000000    1.000000   

          restecg    thalachh        exng     oldpeak         slp         caa  \
count  303.000000  303.000000  303.000000  303.000000  303.000000  303.000000   
mean     0.528053  149.646865    0.326733    1.039604    1.399340    0.729373   
std      0.525860   22.905161    0.469794    1.161075    0.616226    1.022606   
min      0.000000   71.000000    0.000000    0.000000    0.000000    0.000000   
25%      0.000000  133.500000    0.000000    0.000000    1.000000    0.000000   
50%      1.000000  153.000000    0.000000    0.800000    1.000000    0.000000   
75%      1.000000  166.000000    1.000000    1.600000    2.000000    1.000000   
max      2.000000  202.000000    1.000000    6.200000    2.000000    4.000000   

            thall  
count  303.000000  
mean     2.313531  
std      0.612277  
min      0.000000  
25%      2.000000  
50%      2.000000  
75%      3.000000  
max      3.000000

As we can observe from min,max,mean that all the columns are at different scale. So we will Standardize our data as
> Standardization is useful when your data has varying scales and the algorithm you are using does make assumptions about your data having a Gaussian distribution, such as linear regression, logistic regression, and linear discriminant analysis.

we will use Standard Scaler for it

In [12]:
from sklearn.preprocessing import StandardScaler

In [13]:
#let's take all the feature names(column names) into one list.
features=data.columns
ss=StandardScaler()
data_tf=ss.fit_transform(data)
data_tf=pd.DataFrame(data_tf,columns=features)
data_tf.head()

age       sex        cp    trtbps      chol       fbs   restecg  \
0  0.952197  0.681005  1.973123  0.763956 -0.256334  2.394438 -1.005832   
1 -1.915313  0.681005  1.002577 -0.092738  0.072199 -0.417635  0.898962   
2 -1.474158 -1.468418  0.032031 -0.092738 -0.816773 -0.417635 -1.005832   
3  0.180175  0.681005  0.032031 -0.663867 -0.198357 -0.417635  0.898962   
4  0.290464 -1.468418 -0.938515 -0.663867  2.082050 -0.417635  0.898962   

   thalachh      exng   oldpeak       slp       caa     thall  
0  0.015443 -0.696631  1.087338 -2.274579 -0.714429 -2.148873  
1  1.633471 -0.696631  2.122573 -2.274579 -0.714429 -0.512922  
2  0.977514 -0.696631  0.310912  0.976352 -0.714429 -0.512922  
3  1.239897 -0.696631 -0.206705  0.976352 -0.714429 -0.512922  
4  0.583939  1.435481 -0.379244  0.976352 -0.714429 -0.512922

Let's describe our data again

In [14]:
data_tf.describe()

age           sex            cp        trtbps          chol  \
count  3.030000e+02  3.030000e+02  3.030000e+02  3.030000e+02  3.030000e+02   
mean   4.690051e-17 -1.407015e-16  2.345026e-17 -7.035077e-16 -1.113887e-16   
std    1.001654e+00  1.001654e+00  1.001654e+00  1.001654e+00  1.001654e+00   
min   -2.797624e+00 -1.468418e+00 -9.385146e-01 -2.148802e+00 -2.324160e+00   
25%   -7.572802e-01 -1.468418e+00 -9.385146e-01 -6.638668e-01 -6.814943e-01   
50%    6.988599e-02  6.810052e-01  3.203122e-02 -9.273778e-02 -1.210553e-01   
75%    7.316189e-01  6.810052e-01  1.002577e+00  4.783913e-01  5.456738e-01   
max    2.496240e+00  6.810052e-01  1.973123e+00  3.905165e+00  6.140401e+00   

                fbs       restecg      thalachh          exng       oldpeak  \
count  3.030000e+02  3.030000e+02  3.030000e+02  3.030000e+02  3.030000e+02   
mean  -2.345026e-17  1.465641e-16 -6.800574e-16 -4.690051e-17  2.345026e-17   
std    1.001654e+00  1.001654e+00  1.001654e+00  1.001654e+00  1.001654e+00   
min   -4.176345e-01 -1.005832e+00 -3.439267e+00 -6.966305e-01 -8.968617e-01   
25%   -4.176345e-01 -1.005832e+00 -7.061105e-01 -6.966305e-01 -8.968617e-01   
50%   -4.176345e-01  8.989622e-01  1.466343e-01 -6.966305e-01 -2.067053e-01   
75%   -4.176345e-01  8.989622e-01  7.151309e-01  1.435481e+00  4.834512e-01   
max    2.394438e+00  2.803756e+00  2.289429e+00  1.435481e+00  4.451851e+00   

                slp           caa         thall  
count  3.030000e+02  3.030000e+02  3.030000e+02  
mean  -1.407015e-16 -2.345026e-17 -1.641518e-16  
std    1.001654e+00  1.001654e+00  1.001654e+00  
min   -2.274579e+00 -7.144289e-01 -3.784824e+00  
25%   -6.491132e-01 -7.144289e-01 -5.129219e-01  
50%   -6.491132e-01 -7.144289e-01 -5.129219e-01  
75%    9.763521e-01  2.650822e-01  1.123029e+00  
max    9.763521e-01  3.203615e+00  1.123029e+00

Now if we carefully observe our data we will observe that our data is now at same scale

And let's do some analysis on our target columns.

In [15]:
label.value_counts()

1    165
0    138
Name: output, dtype: int64

In [16]:
print('% of 1 label in data set: {ones} %'.format(ones=165*100/(165+138)))
print('% of 0 label in data set: {ones} %'.format(ones=138*100/(165+138)))

% of 1 label in data set: 54.45544554455446 %
% of 0 label in data set: 45.54455445544554 %


we can observe that our data is not much imbalanced as there are **165 datapoints with 1** (54.4 %)label and **138 with 0 label** (45.5%)

In [17]:
from sklearn.model_selection import GridSearchCV
from sklearn.linear_model import LogisticRegression
lr=LogisticRegression(n_jobs=-1,verbose=3,warm_start=True)
param_grid={
    'solver':['lbfgs', 'liblinear', 'newton-cg', 'sag', 'saga'],
    'max_iter':[150,200,100],
}
gs=GridSearchCV(lr,param_grid=param_grid,refit=True,n_jobs=-1,cv=5,scoring='f1',verbose=3,error_score='raise',return_train_score=True)
gs.fit(data_tf,label)

Fitting 5 folds for each of 15 candidates, totalling 75 fits
RUNNING THE L-BFGS-B CODE

           * * *

Machine precision = 2.220D-16
 N =           14     M =           10

At X0         0 variables are exactly at the bounds

At iterate    0    f=  1.67742D+02    |proj g|=  5.80039D+01


ITERATION     1

---------------- CAUCHY entered-------------------
 There are            0   breakpoints 

 GCP found in this segment
Piece      1 --f1, f2 at start point  -2.0021D+04  2.0021D+04
Distance to the stationary point =   1.0000D+00

---------------- exit CAUCHY----------------------

          14  variables are free at GCP            1
 LINE SEARCH           0  times; norm of step =    1.0000000000000000     

At iterate    1    f=  9.63342D+01    |proj g|=  1.57321D+01


ITERATION     2

----------------SUBSM entered-----------------


----------------exit SUBSM --------------------

 LINE SEARCH           0  times; norm of step =   0.26023804463946665     

At iterate    2    f=  9.07

 This problem is unconstrained.
 This problem is unconstrained.
 This problem is unconstrained.
 This problem is unconstrained.
 This problem is unconstrained.


iter  4 act 1.780e-01 pre 1.694e-01 delta 1.149e+00 f 7.596e+01 |g| 2.480e+00 CG   3
iter  5 act 2.105e-03 pre 2.096e-03 delta 1.149e+00 f 7.578e+01 |g| 2.545e-01 CG   3
iter  6 act 7.740e-06 pre 7.739e-06 delta 1.149e+00 f 7.578e+01 |g| 1.678e-02 CG   4
Epoch 1, change: 1.00000000
Epoch 2, change: 0.41035530
Epoch 3, change: 0.46371777
Epoch 4, change: 0.62791315
Epoch 5, change: 0.40433199
Epoch 6, change: 0.20775368
Epoch 7, change: 0.16163068
Epoch 8, change: 0.11550404
Epoch 9, change: 0.09429100
Epoch 10, change: 0.04983609
Epoch 11, change: 0.05424364
Epoch 12, change: 0.01780813
Epoch 13, change: 0.03868383
Epoch 14, change: 0.02110206
Epoch 15, change: 0.01699455
Epoch 16, change: 0.00855553
Epoch 17, change: 0.00434914
Epoch 18, change: 0.00421860
Epoch 19, change: 0.00224540
Epoch 20, change: 0.00270534
Epoch 21, change: 0.00144811
Epoch 22, change: 0.00129614
Epoch 23, change: 0.00077959
Epoch 24, change: 0.00033352
Epoch 25, change: 0.00032368
Epoch 26, change: 0.00023397


 This problem is unconstrained.
 This problem is unconstrained.
 This problem is unconstrained.
 This problem is unconstrained.
 This problem is unconstrained.


iter  2 act 1.240e+01 pre 1.020e+01 delta 1.149e+00 f 9.100e+01 |g| 3.616e+01 CG   2
iter  3 act 2.646e+00 pre 2.317e+00 delta 1.149e+00 f 7.860e+01 |g| 1.143e+01 CG   3
iter  4 act 1.780e-01 pre 1.694e-01 delta 1.149e+00 f 7.596e+01 |g| 2.480e+00 CG   3
iter  5 act 2.105e-03 pre 2.096e-03 delta 1.149e+00 f 7.578e+01 |g| 2.545e-01 CG   3
iter  6 act 7.740e-06 pre 7.739e-06 delta 1.149e+00 f 7.578e+01 |g| 1.678e-02 CG   4


 This problem is unconstrained.
 This problem is unconstrained.
 This problem is unconstrained.
 This problem is unconstrained.
 This problem is unconstrained.


Epoch 1, change: 1.00000000
Epoch 2, change: 0.57082523
Epoch 3, change: 0.38005729
Epoch 4, change: 0.15779527
Epoch 5, change: 0.36009336
Epoch 6, change: 0.17196018
Epoch 7, change: 0.10713111
Epoch 8, change: 0.11428792
Epoch 9, change: 0.04866925
Epoch 10, change: 0.04478725
Epoch 11, change: 0.04855738
Epoch 12, change: 0.02727735
Epoch 13, change: 0.02999638
Epoch 14, change: 0.02418248
Epoch 15, change: 0.01674528
Epoch 16, change: 0.01090395
Epoch 17, change: 0.00510556
Epoch 18, change: 0.00873354
Epoch 19, change: 0.00621897
Epoch 20, change: 0.00744293
Epoch 21, change: 0.00446319
Epoch 22, change: 0.00344088
Epoch 23, change: 0.00202336
Epoch 24, change: 0.00152730
Epoch 25, change: 0.00192293
Epoch 26, change: 0.00062609
Epoch 27, change: 0.00089469
Epoch 28, change: 0.00044436
Epoch 29, change: 0.00037827
Epoch 30, change: 0.00018183
Epoch 31, change: 0.00016027
Epoch 32, change: 0.00010449
Epoch 1, change: 1.00000000
Epoch 2, change: 0.46004156
Epoch 3, change: 0.279393

[Parallel(n_jobs=-1)]: Using backend LokyBackend with 4 concurrent workers.
 This problem is unconstrained.
[Parallel(n_jobs=-1)]: Done   1 out of   1 | elapsed:    0.0s finished


GridSearchCV(cv=5, error_score='raise',
             estimator=LogisticRegression(n_jobs=-1, verbose=3,
                                          warm_start=True),
             n_jobs=-1,
             param_grid={'max_iter': [150, 200, 100],
                         'solver': ['lbfgs', 'liblinear', 'newton-cg', 'sag',
                                    'saga']},
             return_train_score=True, scoring='f1', verbose=3)

In [18]:
gs.cv_results_

{'mean_fit_time': array([0.10665827, 0.00392361, 0.10463829, 0.10457892, 0.10482845,
        0.1058393 , 0.00414147, 0.10539598, 0.1041296 , 0.10502262,
        0.1046813 , 0.00386186, 0.10759711, 0.10449677, 0.10447297]),
 'std_fit_time': array([0.00235486, 0.00093654, 0.00037077, 0.000549  , 0.00064497,
        0.00183041, 0.0013584 , 0.00110354, 0.00042084, 0.00115742,
        0.00129056, 0.00093123, 0.00255565, 0.00038654, 0.00053446]),
 'mean_score_time': array([0.00321326, 0.00258865, 0.00285249, 0.00300679, 0.00358758,
        0.00303664, 0.0024435 , 0.00284877, 0.0028831 , 0.00508552,
        0.00378628, 0.00306168, 0.00305934, 0.00300803, 0.00299959]),
 'std_score_time': array([0.00065889, 0.00038982, 0.00034526, 0.00031284, 0.00107314,
        0.00044269, 0.00042455, 0.00046661, 0.00024674, 0.00416984,
        0.0013871 , 0.00095616, 0.0005317 , 0.00043111, 0.00042244]),
 'param_max_iter': masked_array(data=[150, 150, 150, 150, 150, 200, 200, 200, 200, 200, 100,
             

In [19]:
gs.best_estimator_

LogisticRegression(max_iter=150, n_jobs=-1, verbose=3, warm_start=True)

In [20]:
gs.best_score_

0.8554265609195187

In [21]:
gs.scorer_

make_scorer(f1_score, average=binary)

In [22]:
from sklearn.ensemble import RandomForestClassifier
rfc=RandomForestClassifier(n_jobs=-1,verbose=3)
param_grid={
    'bootstrap':[True,False],
    'class_weight':["balanced", "balanced_subsample"],
    'n_estimators':[100,150,200,250,300,350,400]
}
gs2=GridSearchCV(rfc,param_grid=param_grid,refit=True,n_jobs=-1,cv=5,scoring='f1',verbose=3,error_score='raise',return_train_score=True)
gs2.fit(data_tf,label)

Fitting 5 folds for each of 28 candidates, totalling 140 fits
[CV 2/5] END max_iter=150, solver=lbfgs;, score=(train=0.863, test=0.892) total time=   0.1s
[LibLinear][CV 1/5] END max_iter=150, solver=liblinear;, score=(train=0.880, test=0.845) total time=   0.0s
[LibLinear][CV 2/5] END max_iter=150, solver=liblinear;, score=(train=0.863, test=0.892) total time=   0.0s
[LibLinear][CV 3/5] END max_iter=150, solver=liblinear;, score=(train=0.863, test=0.848) total time=   0.0s
[LibLinear][CV 4/5] END max_iter=150, solver=liblinear;, score=(train=0.862, test=0.886) total time=   0.0s
[CV 1/5] END max_iter=150, solver=newton-cg;, score=(train=0.880, test=0.845) total time=   0.1s
[CV 5/5] END max_iter=150, solver=newton-cg;, score=(train=0.890, test=0.806) total time=   0.1s
convergence after 28 epochs took 0 seconds
[CV 4/5] END max_iter=150, solver=sag;, score=(train=0.862, test=0.886) total time=   0.1s
convergence after 19 epochs took 0 seconds
[CV 3/5] END max_iter=150, solver=saga;, s

[Parallel(n_jobs=-1)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=-1)]: Done   1 out of   1 | elapsed:    0.0s finished
/opt/conda/lib/python3.7/site-packages/sklearn/linear_model/_logistic.py:1526: UserWarning: 'n_jobs' > 1 does not have any effect when 'solver' is set to 'liblinear'. Got 'n_jobs' = 4.
  " = {}.".format(effective_n_jobs(self.n_jobs))
/opt/conda/lib/python3.7/site-packages/sklearn/linear_model/_logistic.py:1526: UserWarning: 'n_jobs' > 1 does not have any effect when 'solver' is set to 'liblinear'. Got 'n_jobs' = 4.
  " = {}.".format(effective_n_jobs(self.n_jobs))
/opt/conda/lib/python3.7/site-packages/sklearn/linear_model/_logistic.py:1526: UserWarning: 'n_jobs' > 1 does not have any effect when 'solver' is set to 'liblinear'. Got 'n_jobs' = 4.
  " = {}.".format(effective_n_jobs(self.n_jobs))
/opt/conda/lib/python3.7/site-packages/sklearn/linear_model/_logistic.py:1526: UserWarning: 'n_jobs' > 1 does not have any effect when 'solver' is


building tree 183 of 250
building tree 184 of 250
building tree 185 of 250
building tree 186 of 250
building tree 187 of 250
building tree 188 of 250
building tree 189 of 250
building tree 190 of 250
building tree 191 of 250
building tree 192 of 250
building tree 193 of 250
building tree 194 of 250
building tree 195 of 250
building tree 196 of 250
building tree 197 of 250
building tree 198 of 250
building tree 199 of 250
building tree 200 of 250
building tree 201 of 250
building tree 202 of 250
building tree 203 of 250
building tree 204 of 250
building tree 205 of 250
building tree 206 of 250
building tree 207 of 250
building tree 208 of 250
building tree 209 of 250
building tree 210 of 250
building tree 211 of 250
building tree 212 of 250
building tree 213 of 250
building tree 214 of 250
building tree 215 of 250
building tree 216 of 250
building tree 217 of 250
building tree 218 of 250
building tree 219 of 250
building tree 220 of 250
building tree 221 of 250
building tree 222 of 250

[Parallel(n_jobs=-1)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=-1)]: Done   1 out of   1 | elapsed:    0.0s finished
/opt/conda/lib/python3.7/site-packages/sklearn/linear_model/_logistic.py:1526: UserWarning: 'n_jobs' > 1 does not have any effect when 'solver' is set to 'liblinear'. Got 'n_jobs' = 4.
  " = {}.".format(effective_n_jobs(self.n_jobs))
[Parallel(n_jobs=-1)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=-1)]: Done   1 out of   1 | elapsed:    0.0s finished
[Parallel(n_jobs=-1)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=-1)]: Done   1 out of   1 | elapsed:    0.0s finished
[Parallel(n_jobs=-1)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=-1)]: Done   1 out of   1 | elapsed:    0.0s finished
[Parallel(n_jobs=-1)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=-1)]: Done   1 out of   1 | elapsed:    0.0s finished
[Paral


building tree 6 of 300
building tree 7 of 300
building tree 8 of 300
building tree 9 of 300
building tree 10 of 300
building tree 11 of 300
building tree 12 of 300
building tree 13 of 300
building tree 14 of 300
building tree 15 of 300
building tree 16 of 300
building tree 17 of 300
building tree 18 of 300
building tree 19 of 300
building tree 20 of 300
building tree 21 of 300
building tree 22 of 300
building tree 23 of 300
building tree 24 of 300
building tree 25 of 300
building tree 26 of 300
building tree 27 of 300
building tree 28 of 300
building tree 29 of 300
building tree 30 of 300
building tree 31 of 300
building tree 32 of 300
building tree 33 of 300
building tree 34 of 300
building tree 35 of 300
building tree 36 of 300
building tree 37 of 300
building tree 38 of 300
building tree 39 of 300
building tree 40 of 300
building tree 41 of 300
building tree 42 of 300
building tree 43 of 300
building tree 44 of 300
building tree 45 of 300
building tree 46 of 300
building tree 47 of

[Parallel(n_jobs=-1)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=-1)]: Done   1 out of   1 | elapsed:    0.0s finished
[Parallel(n_jobs=-1)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=-1)]: Done   1 out of   1 | elapsed:    0.0s finished
[Parallel(n_jobs=-1)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=-1)]: Done   1 out of   1 | elapsed:    0.0s finished
[Parallel(n_jobs=-1)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=-1)]: Done   1 out of   1 | elapsed:    0.0s finished
[Parallel(n_jobs=-1)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=-1)]: Done   1 out of   1 | elapsed:    0.0s finished
[Parallel(n_jobs=-1)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=-1)]: Done   1 out of   1 | elapsed:    0.0s finished
[Parallel(n_jobs=-1)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parall


building tree 261 of 300
building tree 262 of 300
building tree 263 of 300
building tree 264 of 300
building tree 265 of 300
building tree 266 of 300
building tree 267 of 300
building tree 268 of 300
building tree 269 of 300
building tree 270 of 300
building tree 271 of 300
building tree 272 of 300
building tree 273 of 300
building tree 274 of 300
building tree 275 of 300
building tree 276 of 300
building tree 277 of 300
building tree 278 of 300
building tree 279 of 300
building tree 280 of 300
building tree 281 of 300
building tree 282 of 300
building tree 283 of 300
building tree 284 of 300
building tree 285 of 300
building tree 286 of 300
building tree 287 of 300
building tree 288 of 300
building tree 289 of 300
building tree 290 of 300
building tree 291 of 300
building tree 292 of 300
building tree 293 of 300
building tree 294 of 300
building tree 295 of 300
building tree 296 of 300
building tree 297 of 300
building tree 298 of 300
building tree 299 of 300
building tree 300 of 300

[Parallel(n_jobs=-1)]: Done 120 tasks      | elapsed:    0.2s
[Parallel(n_jobs=-1)]: Done 280 tasks      | elapsed:    0.4s
[Parallel(n_jobs=-1)]: Done 300 out of 300 | elapsed:    0.5s finished
[Parallel(n_jobs=4)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=4)]: Done  24 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 120 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 280 tasks      | elapsed:    0.1s
[Parallel(n_jobs=4)]: Done 300 out of 300 | elapsed:    0.1s finished
[Parallel(n_jobs=4)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=4)]: Done  24 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 120 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 280 tasks      | elapsed:    0.1s
[Parallel(n_jobs=4)]: Done 300 out of 300 | elapsed:    0.1s finished
[Parallel(n_jobs=-1)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=-1)]: Done  24 tasks      | elapsed:    0


building tree 59 of 300
building tree 60 of 300
building tree 61 of 300
building tree 62 of 300
building tree 63 of 300
building tree 64 of 300
building tree 65 of 300
building tree 66 of 300
building tree 67 of 300
building tree 68 of 300
building tree 69 of 300
building tree 70 of 300
building tree 71 of 300
building tree 72 of 300
building tree 73 of 300
building tree 74 of 300
building tree 75 of 300
building tree 76 of 300
building tree 77 of 300
building tree 78 of 300
building tree 79 of 300
building tree 80 of 300
building tree 81 of 300
building tree 82 of 300
building tree 83 of 300
building tree 84 of 300
building tree 85 of 300
building tree 86 of 300
building tree 87 of 300
building tree 88 of 300
building tree 89 of 300
building tree 90 of 300
building tree 91 of 300
building tree 92 of 300
building tree 93 of 300
building tree 94 of 300
building tree 95 of 300
building tree 96 of 300
building tree 97 of 300
building tree 98 of 300
building tree 99 of 300
building tree 1

[Parallel(n_jobs=-1)]: Done  24 tasks      | elapsed:    0.1s
[Parallel(n_jobs=-1)]: Done 120 tasks      | elapsed:    0.2s
[Parallel(n_jobs=-1)]: Done 280 tasks      | elapsed:    0.4s
[Parallel(n_jobs=-1)]: Done 300 out of 300 | elapsed:    0.4s finished
[Parallel(n_jobs=4)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=4)]: Done  24 tasks      | elapsed:    0.1s
[Parallel(n_jobs=4)]: Done 120 tasks      | elapsed:    0.1s
[Parallel(n_jobs=4)]: Done 280 tasks      | elapsed:    0.1s
[Parallel(n_jobs=4)]: Done 300 out of 300 | elapsed:    0.1s finished
[Parallel(n_jobs=4)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=4)]: Done  24 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 120 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 280 tasks      | elapsed:    0.1s
[Parallel(n_jobs=4)]: Done 300 out of 300 | elapsed:    0.1s finished
[Parallel(n_jobs=-1)]: Using backend ThreadingBackend with 4 concurrent worke

building tree 179 of 300
building tree 180 of 300
building tree 181 of 300

building tree 182 of 300
building tree 183 of 300
building tree 184 of 300
building tree 185 of 300
building tree 186 of 300
building tree 187 of 300
building tree 188 of 300
building tree 189 of 300
building tree 190 of 300
building tree 191 of 300
building tree 192 of 300
building tree 193 of 300
building tree 194 of 300
building tree 195 of 300
building tree 196 of 300
building tree 197 of 300
building tree 198 of 300
building tree 199 of 300
building tree 200 of 300
building tree 201 of 300
building tree 202 of 300
building tree 203 of 300
building tree 204 of 300
building tree 205 of 300
building tree 206 of 300
building tree 207 of 300
building tree 208 of 300
building tree 209 of 300
building tree 210 of 300
building tree 211 of 300
building tree 212 of 300
building tree 213 of 300
building tree 214 of 300
building tree 215 of 300
building tree 216 of 300
building tree 217 of 300
building tree 218 of 300

[Parallel(n_jobs=4)]: Done 280 tasks      | elapsed:    0.1s
[Parallel(n_jobs=4)]: Done 350 out of 350 | elapsed:    0.1s finished
[Parallel(n_jobs=4)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=4)]: Done  24 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 120 tasks      | elapsed:    0.1s
[Parallel(n_jobs=4)]: Done 280 tasks      | elapsed:    0.1s
[Parallel(n_jobs=4)]: Done 350 out of 350 | elapsed:    0.1s finished
[Parallel(n_jobs=-1)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=-1)]: Done  24 tasks      | elapsed:    0.1s
[Parallel(n_jobs=-1)]: Done 120 tasks      | elapsed:    0.2s
[Parallel(n_jobs=-1)]: Done 280 tasks      | elapsed:    0.4s
[Parallel(n_jobs=-1)]: Done 400 out of 400 | elapsed:    0.6s finished
[Parallel(n_jobs=4)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=4)]: Done  24 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 120 tasks      | elapsed:    0


building tree 215 of 350
building tree 216 of 350
building tree 217 of 350
building tree 218 of 350
building tree 219 of 350
building tree 220 of 350
building tree 221 of 350
building tree 222 of 350
building tree 223 of 350
building tree 224 of 350
building tree 225 of 350
building tree 226 of 350
building tree 227 of 350
building tree 228 of 350
building tree 229 of 350
building tree 230 of 350
building tree 231 of 350
building tree 232 of 350
building tree 233 of 350
building tree 234 of 350
building tree 235 of 350
building tree 236 of 350
building tree 237 of 350
building tree 238 of 350
building tree 239 of 350
building tree 240 of 350
building tree 241 of 350
building tree 242 of 350
building tree 243 of 350
building tree 244 of 350
building tree 245 of 350
building tree 246 of 350
building tree 247 of 350
building tree 248 of 350
building tree 249 of 350
building tree 250 of 350
building tree 251 of 350
building tree 252 of 350
building tree 253 of 350
building tree 254 of 350

[Parallel(n_jobs=-1)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=-1)]: Done  24 tasks      | elapsed:    0.1s
[Parallel(n_jobs=-1)]: Done 120 tasks      | elapsed:    0.2s
[Parallel(n_jobs=-1)]: Done 280 tasks      | elapsed:    0.4s
[Parallel(n_jobs=-1)]: Done 350 out of 350 | elapsed:    0.5s finished
[Parallel(n_jobs=4)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=4)]: Done  24 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 120 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 280 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 350 out of 350 | elapsed:    0.1s finished
[Parallel(n_jobs=4)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=4)]: Done  24 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 120 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 280 tasks      | elapsed:    0.1s
[Parallel(n_jobs=4)]: Done 350 out of 350 | elapsed:    0.1s finis


building tree 63 of 400
building tree 64 of 400
building tree 65 of 400
building tree 66 of 400
building tree 67 of 400
building tree 68 of 400
building tree 69 of 400
building tree 70 of 400
building tree 71 of 400
building tree 72 of 400
building tree 73 of 400
building tree 74 of 400
building tree 75 of 400
building tree 76 of 400
building tree 77 of 400
building tree 78 of 400
building tree 79 of 400
building tree 80 of 400
building tree 81 of 400
building tree 82 of 400
building tree 83 of 400
building tree 84 of 400
building tree 85 of 400
building tree 86 of 400
building tree 87 of 400
building tree 88 of 400
building tree 89 of 400
building tree 90 of 400
building tree 91 of 400
building tree 92 of 400
building tree 93 of 400
building tree 94 of 400
building tree 95 of 400
building tree 96 of 400
building tree 97 of 400
building tree 98 of 400
building tree 99 of 400
building tree 100 of 400
building tree 101 of 400
building tree 102 of 400
building tree 103 of 400
building tr

[Parallel(n_jobs=4)]: Done 120 tasks      | elapsed:    0.1s
[Parallel(n_jobs=4)]: Done 280 tasks      | elapsed:    0.1s
[Parallel(n_jobs=4)]: Done 300 out of 300 | elapsed:    0.1s finished
[Parallel(n_jobs=-1)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=-1)]: Done  24 tasks      | elapsed:    0.1s
[Parallel(n_jobs=-1)]: Done 120 tasks      | elapsed:    0.2s
[Parallel(n_jobs=-1)]: Done 280 tasks      | elapsed:    0.5s
[Parallel(n_jobs=-1)]: Done 350 out of 350 | elapsed:    0.7s finished
[Parallel(n_jobs=4)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=4)]: Done  24 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 120 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 280 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 350 out of 350 | elapsed:    0.1s finished
[Parallel(n_jobs=4)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=4)]: Done  24 tasks      | elapsed:    0


building tree 1 of 300
building tree 2 of 300
building tree 3 of 300
building tree 4 of 300
building tree 5 of 300
building tree 6 of 300
building tree 7 of 300
building tree 8 of 300
building tree 9 of 300
building tree 10 of 300
building tree 11 of 300
building tree 12 of 300
building tree 13 of 300
building tree 14 of 300
building tree 15 of 300
building tree 16 of 300
building tree 17 of 300
building tree 18 of 300
building tree 19 of 300
building tree 20 of 300
building tree 21 of 300
building tree 22 of 300
building tree 23 of 300
building tree 24 of 300
building tree 25 of 300
building tree 26 of 300
building tree 27 of 300
building tree 28 of 300
building tree 29 of 300
building tree 30 of 300
building tree 31 of 300
building tree 32 of 300
building tree 33 of 300
building tree 34 of 300
building tree 35 of 300
building tree 36 of 300
building tree 37 of 300
building tree 38 of 300
building tree 39 of 300
building tree 40 of 300
building tree 41 of 300
building tree 42 of 300


[Parallel(n_jobs=4)]: Done  24 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 120 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 280 tasks      | elapsed:    0.1s
[Parallel(n_jobs=4)]: Done 350 out of 350 | elapsed:    0.1s finished
[Parallel(n_jobs=4)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=4)]: Done  24 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 120 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 280 tasks      | elapsed:    0.1s
[Parallel(n_jobs=4)]: Done 350 out of 350 | elapsed:    0.1s finished
[Parallel(n_jobs=-1)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=-1)]: Done  24 tasks      | elapsed:    0.1s
[Parallel(n_jobs=-1)]: Done 120 tasks      | elapsed:    0.2s
[Parallel(n_jobs=-1)]: Done 280 tasks      | elapsed:    0.6s
[Parallel(n_jobs=-1)]: Done 350 out of 350 | elapsed:    0.7s finished
[Parallel(n_jobs=4)]: Using backend ThreadingBackend with 4 concurrent worke


building tree 112 of 350
building tree 113 of 350
building tree 114 of 350
building tree 115 of 350
building tree 116 of 350
building tree 117 of 350
building tree 118 of 350
building tree 119 of 350
building tree 120 of 350
building tree 121 of 350
building tree 122 of 350
building tree 123 of 350
building tree 124 of 350
building tree 125 of 350
building tree 126 of 350
building tree 127 of 350
building tree 128 of 350
building tree 129 of 350
building tree 130 of 350
building tree 131 of 350
building tree 132 of 350
building tree 133 of 350
building tree 134 of 350
building tree 135 of 350
building tree 136 of 350
building tree 137 of 350
building tree 138 of 350
building tree 139 of 350
building tree 140 of 350
building tree 141 of 350
building tree 142 of 350
building tree 143 of 350
building tree 144 of 350
building tree 145 of 350
building tree 146 of 350
building tree 147 of 350
building tree 148 of 350
building tree 149 of 350
building tree 150 of 350
building tree 151 of 350

[Parallel(n_jobs=4)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=4)]: Done  24 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 120 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 280 tasks      | elapsed:    0.1s
[Parallel(n_jobs=4)]: Done 350 out of 350 | elapsed:    0.1s finished
[Parallel(n_jobs=-1)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=-1)]: Done  24 tasks      | elapsed:    0.1s
[Parallel(n_jobs=-1)]: Done 120 tasks      | elapsed:    0.3s
[Parallel(n_jobs=-1)]: Done 280 tasks      | elapsed:    0.6s
[Parallel(n_jobs=-1)]: Done 400 out of 400 | elapsed:    0.8s finished
[Parallel(n_jobs=4)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=4)]: Done  24 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 120 tasks      | elapsed:    0.1s
[Parallel(n_jobs=4)]: Done 280 tasks      | elapsed:    0.1s
[Parallel(n_jobs=4)]: Done 400 out of 400 | elapsed:    0.1s finis


building tree 89 of 350
building tree 90 of 350
building tree 91 of 350
building tree 92 of 350
building tree 93 of 350
building tree 94 of 350
building tree 95 of 350
building tree 96 of 350
building tree 97 of 350
building tree 98 of 350
building tree 99 of 350
building tree 100 of 350
building tree 101 of 350
building tree 102 of 350
building tree 103 of 350
building tree 104 of 350
building tree 105 of 350
building tree 106 of 350
building tree 107 of 350
building tree 108 of 350
building tree 109 of 350
building tree 110 of 350
building tree 111 of 350
building tree 112 of 350
building tree 113 of 350
building tree 114 of 350
building tree 115 of 350
building tree 116 of 350
building tree 117 of 350
building tree 118 of 350
building tree 119 of 350
building tree 120 of 350
building tree 121 of 350
building tree 122 of 350
building tree 123 of 350
building tree 124 of 350
building tree 125 of 350
building tree 126 of 350
building tree 127 of 350
building tree 128 of 350
building t

[Parallel(n_jobs=-1)]: Done 400 out of 400 | elapsed:    0.8s finished
[Parallel(n_jobs=4)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=4)]: Done  24 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 120 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 280 tasks      | elapsed:    0.1s
[Parallel(n_jobs=4)]: Done 400 out of 400 | elapsed:    0.1s finished
[Parallel(n_jobs=4)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=4)]: Done  24 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 120 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 280 tasks      | elapsed:    0.1s
[Parallel(n_jobs=4)]: Done 400 out of 400 | elapsed:    0.1s finished
[Parallel(n_jobs=-1)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=-1)]: Done  24 tasks      | elapsed:    0.1s
[Parallel(n_jobs=-1)]: Done 120 tasks      | elapsed:    0.3s
[Parallel(n_jobs=-1)]: Done 280 tasks      | elapsed:    0


building tree 21 of 400
building tree 22 of 400
building tree 23 of 400
building tree 24 of 400
building tree 25 of 400
building tree 26 of 400
building tree 27 of 400
building tree 28 of 400
building tree 29 of 400
building tree 30 of 400
building tree 31 of 400
building tree 32 of 400
building tree 33 of 400
building tree 34 of 400
building tree 35 of 400
building tree 36 of 400
building tree 37 of 400
building tree 38 of 400
building tree 39 of 400
building tree 40 of 400
building tree 41 of 400
building tree 42 of 400
building tree 43 of 400
building tree 44 of 400
building tree 45 of 400
building tree 46 of 400
building tree 47 of 400
building tree 48 of 400
building tree 49 of 400
building tree 50 of 400
building tree 51 of 400
building tree 52 of 400
building tree 53 of 400
building tree 54 of 400
building tree 55 of 400
building tree 56 of 400
building tree 57 of 400
building tree 58 of 400
building tree 59 of 400
building tree 60 of 400
building tree 61 of 400
building tree 6

[Parallel(n_jobs=4)]: Done 120 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 280 tasks      | elapsed:    0.1s
[Parallel(n_jobs=4)]: Done 350 out of 350 | elapsed:    0.1s finished
[Parallel(n_jobs=4)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=4)]: Done  24 tasks      | elapsed:    0.1s
[Parallel(n_jobs=4)]: Done 120 tasks      | elapsed:    0.1s
[Parallel(n_jobs=4)]: Done 280 tasks      | elapsed:    0.1s
[Parallel(n_jobs=4)]: Done 350 out of 350 | elapsed:    0.1s finished
[Parallel(n_jobs=-1)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=-1)]: Done  24 tasks      | elapsed:    0.0s
[Parallel(n_jobs=-1)]: Done 120 tasks      | elapsed:    0.2s
[Parallel(n_jobs=-1)]: Done 280 tasks      | elapsed:    0.4s
[Parallel(n_jobs=-1)]: Done 350 out of 350 | elapsed:    0.5s finished
[Parallel(n_jobs=4)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=4)]: Done  24 tasks      | elapsed:    0


building tree 42 of 350
building tree 43 of 350
building tree 44 of 350
building tree 45 of 350
building tree 46 of 350
building tree 47 of 350
building tree 48 of 350
building tree 49 of 350
building tree 50 of 350
building tree 51 of 350
building tree 52 of 350
building tree 53 of 350
building tree 54 of 350
building tree 55 of 350
building tree 56 of 350
building tree 57 of 350
building tree 58 of 350
building tree 59 of 350
building tree 60 of 350
building tree 61 of 350
building tree 62 of 350
building tree 63 of 350
building tree 64 of 350
building tree 65 of 350
building tree 66 of 350
building tree 67 of 350
building tree 68 of 350
building tree 69 of 350
building tree 70 of 350
building tree 71 of 350
building tree 72 of 350
building tree 73 of 350
building tree 74 of 350
building tree 75 of 350
building tree 76 of 350
building tree 77 of 350
building tree 78 of 350
building tree 79 of 350
building tree 80 of 350
building tree 81 of 350
building tree 82 of 350
building tree 8

[Parallel(n_jobs=4)]: Done  24 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 120 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 280 tasks      | elapsed:    0.1s
[Parallel(n_jobs=4)]: Done 350 out of 350 | elapsed:    0.1s finished
[Parallel(n_jobs=-1)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=-1)]: Done  24 tasks      | elapsed:    0.1s
[Parallel(n_jobs=-1)]: Done 120 tasks      | elapsed:    0.2s
[Parallel(n_jobs=-1)]: Done 280 tasks      | elapsed:    0.3s
[Parallel(n_jobs=-1)]: Done 400 out of 400 | elapsed:    0.5s finished
[Parallel(n_jobs=4)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=4)]: Done  24 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 120 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 280 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 400 out of 400 | elapsed:    0.1s finished
[Parallel(n_jobs=4)]: Using backend ThreadingBackend with 4 concurrent worke


building tree 49 of 350
building tree 50 of 350
building tree 51 of 350
building tree 52 of 350
building tree 53 of 350
building tree 54 of 350
building tree 55 of 350
building tree 56 of 350
building tree 57 of 350
building tree 58 of 350
building tree 59 of 350
building tree 60 of 350
building tree 61 of 350
building tree 62 of 350
building tree 63 of 350
building tree 64 of 350
building tree 65 of 350
building tree 66 of 350
building tree 67 of 350
building tree 68 of 350
building tree 69 of 350
building tree 70 of 350
building tree 71 of 350
building tree 72 of 350
building tree 73 of 350
building tree 74 of 350
building tree 75 of 350
building tree 76 of 350
building tree 77 of 350
building tree 78 of 350
building tree 79 of 350
building tree 80 of 350
building tree 81 of 350
building tree 82 of 350
building tree 83 of 350
building tree 84 of 350
building tree 85 of 350
building tree 86 of 350
building tree 87 of 350
building tree 88 of 350
building tree 89 of 350
building tree 9

[Parallel(n_jobs=4)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=4)]: Done  24 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 120 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 280 tasks      | elapsed:    0.1s
[Parallel(n_jobs=4)]: Done 400 out of 400 | elapsed:    0.1s finished
[Parallel(n_jobs=4)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=4)]: Done  24 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 120 tasks      | elapsed:    0.0s
[Parallel(n_jobs=4)]: Done 280 tasks      | elapsed:    0.1s
[Parallel(n_jobs=4)]: Done 400 out of 400 | elapsed:    0.1s finished
[Parallel(n_jobs=-1)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=-1)]: Done  24 tasks      | elapsed:    0.0s
[Parallel(n_jobs=-1)]: Done 120 tasks      | elapsed:    0.2s
[Parallel(n_jobs=-1)]: Done 280 tasks      | elapsed:    0.4s
[Parallel(n_jobs=-1)]: Done 400 out of 400 | elapsed:    0.5s finis


building tree 70 of 400
building tree 71 of 400
building tree 72 of 400
building tree 73 of 400
building tree 74 of 400
building tree 75 of 400
building tree 76 of 400
building tree 77 of 400
building tree 78 of 400
building tree 79 of 400
building tree 80 of 400
building tree 81 of 400
building tree 82 of 400
building tree 83 of 400
building tree 84 of 400
building tree 85 of 400
building tree 86 of 400
building tree 87 of 400
building tree 88 of 400
building tree 89 of 400
building tree 90 of 400
building tree 91 of 400
building tree 92 of 400
building tree 93 of 400
building tree 94 of 400
building tree 95 of 400
building tree 96 of 400
building tree 97 of 400
building tree 98 of 400
building tree 99 of 400
building tree 100 of 400
building tree 101 of 400
building tree 102 of 400
building tree 103 of 400
building tree 104 of 400
building tree 105 of 400
building tree 106 of 400
building tree 107 of 400
building tree 108 of 400
building tree 109 of 400
building tree 110 of 400
buil

[Parallel(n_jobs=-1)]: Using backend ThreadingBackend with 4 concurrent workers.
[Parallel(n_jobs=-1)]: Done  24 tasks      | elapsed:    0.1s


building tree 1 of 150
building tree 2 of 150
building tree 3 of 150
building tree 4 of 150
building tree 5 of 150
building tree 6 of 150
building tree 7 of 150
building tree 8 of 150
building tree 9 of 150
building tree 10 of 150
building tree 11 of 150
building tree 12 of 150
building tree 13 of 150
building tree 14 of 150
building tree 15 of 150
building tree 16 of 150
building tree 17 of 150
building tree 18 of 150
building tree 19 of 150
building tree 20 of 150
building tree 21 of 150
building tree 22 of 150
building tree 23 of 150
building tree 24 of 150
building tree 25 of 150
building tree 26 of 150
building tree 27 of 150
building tree 28 of 150
building tree 29 of 150
building tree 30 of 150
building tree 31 of 150
building tree 32 of 150
building tree 33 of 150
building tree 34 of 150
building tree 35 of 150
building tree 36 of 150
building tree 37 of 150
building tree 38 of 150
building tree 39 of 150
building tree 40 of 150
building tree 41 of 150
building tree 42 of 150
b

[Parallel(n_jobs=-1)]: Done 120 tasks      | elapsed:    0.3s
[Parallel(n_jobs=-1)]: Done 150 out of 150 | elapsed:    0.4s finished


GridSearchCV(cv=5, error_score='raise',
             estimator=RandomForestClassifier(n_jobs=-1, verbose=3), n_jobs=-1,
             param_grid={'bootstrap': [True, False],
                         'class_weight': ['balanced', 'balanced_subsample'],
                         'n_estimators': [100, 150, 200, 250, 300, 350, 400]},
             return_train_score=True, scoring='f1', verbose=3)

In [23]:
gs2.best_estimator_

RandomForestClassifier(class_weight='balanced_subsample', n_estimators=150,
                       n_jobs=-1, verbose=3)

In [24]:
gs2.best_score_

0.8501630434782609

**We can see here that our LogisticRegression worked better then RandomForestClassifier
so for further deployment task we will be using our LogisticRegression Model**
> **Deployment will be uploaded on my github profile**